templates/flavors/rke2/vlan/kustomization.yaml

```
@@ -0,0 +1,148 @@
+      kind: MachineDeployment
+      metadata:
+        name: ${CLUSTER_NAME}-md-0
+        annotations:
+          cluster.x-k8s.io/cluster-api-autoscaler-node-group-min-size: ${WORKER_MACHINE_MIN:-"1"}
+          cluster.x-k8s.io/cluster-api-autoscaler-node-group-max-size: ${WORKER_MACHINE_MAX:-"10"}
+  - target:
+      group: infrastructure.cluster.x-k8s.io
+      version: v1alpha2
+      kind: LinodeMachineTemplate
+      name: .*-control-plane
+    patch: |-
+      apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+      kind: LinodeMachineTemplate
+      metadata:
+        name: ${CLUSTER_NAME}-control-plane
+      spec:
+        template:
+          spec:
+            dataDisks:
+              sdc:
+                label: etcd_disk
+                size: 10Gi
+  - target:
+      group: controlplane.cluster.x-k8s.io
+      version: v1beta1
+      kind: RKE2ControlPlane
+    patch: |-
+      apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+      kind: RKE2ControlPlane
+      metadata:
+        name: ${CLUSTER_NAME}-control-plane
+      spec:
+        agentConfig:
+          additionalUserData:
+            config: |-
+              diskSetup:
+                /dev/sdc:
+                  layout: true
+              fs_setup:
+                - label: /dev/sdc
+                  filesystem: ext4
+                  device: /dev/sdc
+              mounts:
+                - ["/dev/sdc", "/var/lib/etcd_data"]
+  - target:
+      group: controlplane.cluster.x-k8s.io
+      version: v1beta1
+      kind: RKE2ControlPlane
+    patch: |-
+      - op: replace
+        path: /spec/preRKE2Commands
+        value:
+          - |
+            mkdir -p /etc/rancher/rke2/config.yaml.d/
+            echo "node-ip: $(hostname -I | grep -oE 10\.0\.[0-9]+\.[0-9]+)" >> /etc/rancher/rke2/config.yaml.d/capi-config.yaml
+          - sed -i '/swap/d' /etc/fstab
+          - swapoff -a
+          - hostnamectl set-hostname '{{ ds.meta_data.label }}' && hostname -F /etc/hostname
+  - target:
+      group: infrastructure.cluster.x-k8s.io
+      version: v1alpha2
+      kind: LinodeMachineTemplate
+      name: .*-control-plane
+    patch: |-
+      apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+      kind: LinodeMachineTemplate
+      metadata:
+        name: ${CLUSTER_NAME}-control-plane
+      spec:
+        template:
+          spec:
+            placementGroupRef:
+              apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+              kind: LinodePlacementGroup
+              name: ${CLUSTER_NAME}
+  - target:
+      group: bootstrap.cluster.x-k8s.io
+      version: v1beta1
+      kind: RKE2ConfigTemplate
+    patch: |-
+      - op: replace
+        path: /spec/template/spec/preRKE2Commands
+        value:
+          - |
+            mkdir -p /etc/rancher/rke2/config.yaml.d/
+            echo "node-ip: $(hostname -I | grep -oE 10\.0\.[0-9]+\.[0-9]+)" >> /etc/rancher/rke2/config.yaml.d/capi-config.yaml
```

```suggestion
            echo "node-ip: $(hostname -I | grep -oE ^10\.0\.[0-9]+\.[0-9]+)" >> /etc/rancher/rke2/config.yaml.d/capi-config.yaml
```
avoids an edge case where extra IPs get matched that shouldn't